server/src/main/java/org/opensearch/action/admin/indices/segments/IndicesSegmentResponse.java

```diff
@@ -87,20 +85,19 @@ public Map<String, IndexSegments> getIndices() {
         }
         Map<String, IndexSegments> indicesSegments = new HashMap<>();
 
-        Set<String> indices = new HashSet<>();
+        Map<String, List<ShardSegments>> indicesShardSegments = new HashMap<>();
```

Comment:
Wouldn't it now consume more memory - given that the map is retained for the lifetime of the method ?